codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerInstantiatorTest.kt

```diff
@@ -233,4 +242,98 @@ class ServerInstantiatorTest {
         }
         project.compileAndTest()
     }
+
+    @Test
+    fun `uses writable for shapes`() {
+        val inner = model.lookup<StructureShape>("com.test#Inner")
+        val nestedStruct = model.lookup<StructureShape>("com.test#NestedStruct")
+        val sut = ServerInstantiator(codegenContext)
+
+        val project = TestWorkspace.testProject(model)
+        inner.renderWithModelBuilder(model, symbolProvider, project)
+        nestedStruct.renderWithModelBuilder(model, symbolProvider, project)
+        project.moduleFor(inner) {
+            val nestedUnion = model.lookup<UnionShape>("com.test#NestedUnion")
+            UnionGenerator(model, symbolProvider, this, nestedUnion).render()
+
+            unitTest("writable_for_maps") {
```

Comment:
Not sure I follow why this test needs to be so complex (maps within maps) to test that we're interpolating a very simple writable `Some(result)`.